cql3/statements/alter_keyspace_statement.cc

```diff
@@ -55,6 +68,17 @@ void cql3::statements::alter_keyspace_statement::validate(query_processor& qp, c
         }
         try {
             auto ks = qp.db().find_keyspace(_name);
+
+            if (ks.get_replication_strategy().uses_tablets()) {
+                const std::map<sstring, sstring>& current_rfs = ks.metadata()->strategy_options();
+                for (const auto& [new_dc, new_rf] : _attrs->get_replication_options()) {
+                    auto it = current_rfs.find(new_dc);
+                    if (it != current_rfs.end() && !validate_rf_difference(it->second, new_rf)) {
```

Comment:
1. So we allow new_rf to be greater than 1 for new DCs?
(in their case, there could be no value in the current strategy_options)

2. How about the `replication_factor` option?
It seems like it's being handled same as per-dc options.
Not sure that it needs to be validated the same way as it applies only to new DCs.
Initially it might have been set to e.g. 3, but for a new DC, to comply with the +/-1 constraint, one would pass `{`replication_factor`: 1}` and than `3 - 1 = 2` that is `> 1`.
I wish we never supported this abomination of config options, maybe it's a good opportunity to not support it for tablets.

Cc @xemul @nyh

Reply:
> 2. How about the replication_factor option?

For NetworkTopologyStrategy this option shouldn't be messed with here. The as_ks_metadata_update() is called below, it will replace this option with per-dc options, so the whole check should be moved below

Reply:
> So maybe it's a good opportunity to not support it at all for tablets, neither for CREATE KEYSPACE nor for ALTER KEYSPACE.

As discussed previously -- for CREATE KEYSPACE the option is very convenient as it lets user start working without even knowing the DC name (e.g. -- by copy-n-paste-ing examples from tutorials/docs/whatever).

For ALTER KEYSPACE it can be a good opportunity to deprecate this short-cut indeed, but I think it's better to be more delicate here. If there's only one DC in the configuration, then take the `replication_factor` option as a universal alias for it, but if there are many, then do nothing and return an error.

Reply:
> So we allow new_rf to be greater than 1 for new DCs?

Yes, but it's going to be handled in the PR that handled adding new DCs with tablets enabled.

> How about the replication_factor option

We're rejecting ALTERs with `replication_factor` tag.